File: src/test/java/eu/bitwalker/useragentutils/VersionTest.java
package eu.bitwalker.useragentutils;

import static org.junit.Assert.*;

import org.junit.Test;

public class VersionTest {

    @Test
    public void shouldCompareCorrectly() {
        assertTrue(new Version("2.5", "2", "5").compareTo(new Version("3.1", "3", "1")) < 0);
        assertTrue(new Version("7.0.1", "7", "0.1").compareTo(new Version("7.0", "7", "0")) > 0);
        assertTrue(new Version("7.0.1", "7", "0.1").compareTo(new Version("7.0.1", "7", "0.1")) == 0);
        assertTrue(new Version("7.0.a", "7", "0.a").compareTo(new Version("7.0.b", "7", "0.b")) < 0);
        assertTrue(new Version("25.0", "25", "0").compareTo(new Version("3.1", "3", "1")) > 0);
    }
}

File: src/test/csharp/eu/bitwalker/useragentutils/VersionTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace eu.bitwalker.useragentutils
{
    [TestClass]
    public class VersionTest
    {

        [TestMethod]
        public void shouldCompareCorrectly()
        {
            Assert.IsTrue(new Version("2.5", "2", "5").CompareTo(new Version("3.1", "3", "1")) < 0);
            Assert.IsTrue(new Version("7.0.1", "7", "0.1").CompareTo(new Version("7.0", "7", "0")) > 0);
            Assert.IsTrue(new Version("7.0.1", "7", "0.1").CompareTo(new Version("7.0.1", "7", "0.1")) == 0);
            Assert.IsTrue(new Version("7.0.a", "7", "0.a").CompareTo(new Version("7.0.b", "7", "0.b")) < 0);
            Assert.IsTrue(new Version("25.0", "25", "0").CompareTo(new Version("3.1", "3", "1")) > 0);
        }
    }
}